@re-/assert/test/metaTests/testHelpers.ts

```
@@ -0,0 +1,110 @@
+    functionName: string,
+    varStatements: VariableStatement[]
+) =>
+    varStatements
+        .find((statement) => statement.getText().includes(functionName))
+        ?.getDescendantsOfKind(SyntaxKind.ExpressionStatement)
+
+const getTestComment = (statement: ExpressionStatement, text: string) => {
+    const range = statement.getLeadingCommentRanges()
+    return text
+        .slice(range[0].getPos(), range[0].getEnd())
+        .replace("//", "")
+        .trim()
+}
+export const cleanUpSourceFile = async (sf: SourceFile, template: string) => {
+    sf.replaceWithText(template)
+    await sf.save()
+}
+export const getExpectedResult = (
+    statement: ExpressionStatement,
+    expectedData: Record<string, string>,
+    text: string
+) => expectedData[getTestComment(statement, text)]
+
+export const checkIfTestsContainErrors = (
+    statements: ExpressionStatement[],
+    expectedValues: Record<string, string>,
+    fullText: string
+) => {
+    let hasErrors = false
+    for (const statement of statements) {
+        const expected = getExpectedResult(statement, expectedValues, fullText)
+        try {
+            strict.deepEqual(statement.getText(), expected)
+        } catch (e) {
+            // @ts-ignore
+            console.log(`actual: ${e.actual} \nexpected: ${e.expected}`)
+            hasErrors = true
+            break
+        }
+    }
+    return hasErrors
+}
+export const checkThatBenchSnapGetsPopulated = (
+    statements: ExpressionStatement[]
+) => {
+    const meanMedianRegex = /\.(median|mean)\(`\d+\.\d+(ns|us|s|ms)`\)/
+    const markRegex =
+        /\.mark\(\{(mean|median): ?`\d+\.+\d+(ns|us|ms|s)\\, ?median: ?`\d+\.\d+(ns|us|ms|s)`\}\)/
+    let errors = false
+    for (const statement of statements) {
+        if (
+            !(
+                meanMedianRegex.test(statement.getText()) ||
+                markRegex.test(statement.getText())
+            )
+        ) {
+            errors = true
```

Here you should also just throw an assertion error directly like:

```ts
throw new AssertionError(`${statement.getText()} did not match the expected benchmark pattern.`)
```